sls-versions/src/main/java/com/palantir/sls/versions/CompactVersion.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.sls.versions;
+
+import com.google.errorprone.annotations.CompileTimeConstant;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import java.util.OptionalInt;
+
+/**
+ * Stores a compact representation of {@link OrderableSlsVersion} that is lexicographically ordered when rendered as
+ * a sequence of bytes. CompactVersion allocates 20 bits to store each of the five numeric components of an
+ * {@link OrderableSlsVersion}. To enable representing bytes as two SafeLong values, this implementation packs at most
+ * 53 bits into a single {@code long}.
+ *
+ * <p>Bits are allocated as follows, from lowest bits to highest:
+ * <code>
+ * LSB SafeLong:
+ * 20 bits: distance from release
+ *  2 bits: priority1 (1 = RELEASE_CANDIDATE_SNAPSHOT, 0 = else)
+ * 20 bits: RC number
+ *  2 bits: priority2 (2 = RELEASE_SNAPSHOT, 1 = RELEASE, 0 = RELEASE_CANDIDATE)
```

Comment:
3=?

Reply:
3 is an unused value

Reply:
noted unused in docs, and same for unused `priority1` values.